mcmetadata/__init__.py

```diff
@@ -119,6 +119,15 @@ def extract(url: str, html_text: Optional[str] = None, include_other_metadata: O
     language_duration = time.monotonic() - t1
     stats_accumulator['language'] += language_duration
 
+    # canonical url
+    t1 = time.monotonic()
+    if 'canonical_url' in overrides:
+        canonical_url = overrides['canonical_url']
+    else:
+        canonical_url = article.get('canonical_url')
+    canonical_url_duration = time.monotonic() - t1
```

Comment:
There's no real work (separate from the "content" step) being done, so I don't think it needs to have its own STAT/timing data!